CRM/Core/BAO/CustomField.php

```diff
@@ -183,13 +183,13 @@ public static function create($params) {
       // make sure all values are present in the object
       $customField->find(TRUE);
 
-      self::createField($customField, 'add', FALSE, $triggerRebuild);
+      self::createField($customField, $op, FALSE, $triggerRebuild);
     }
 
     // complete transaction
     $transaction->commit();
 
-    CRM_Utils_Hook::post($op, 'CustomField', $customField->id, $customField);
+    CRM_Utils_Hook::post(($op === 'add' ? 'create' : 'edit'), 'CustomField', $customField->id, $customField);
```

Comment:
Can we use same concept here?
`empty($params['id']) ? 'create' : 'edit';`

Since hook_pre() allows once to change the $params and its possible that one can set  $params['id'](very rare case though)

Reply:
@pradpnayak I thought about your comment here but in other BAO the op is determined before calling the 'pre' hook & retained for the post hook so that would be inconsistent with other BAO. I am going to add some code comments but to a subsequent PR as they need rebasing after the merges that have been done so far